location-service/src/main/java/com/movie/recommender/location/controller/LocationController.java

```diff
@@ -17,14 +19,17 @@
 public class LocationController {
 
     private final LocationService locationService;
+    private final RabbitTemplate rabbitTemplate;
 
-    public LocationController(LocationService locationService) {
+    public LocationController(LocationService locationService, RabbitTemplate rabbitTemplate) {
         this.locationService = locationService;
+        this.rabbitTemplate = rabbitTemplate;
     }
 
     @PostMapping
     public ResponseEntity<LocationDTO> createLocation(@RequestBody @Valid LocationCreateDTO locationCreateDTO) {
         LocationDTO savedLocation = locationService.saveLocation(locationCreateDTO);
+        triggerCrawlProcess("create");
```

Comment:
send the message only when a new city is added to the db
crawling is a time consuming and expensive operation, so it cannot be triggered for every location creation or update
add this logic to `LocationService`, not to the controller
in `LocationService`, the method `findOrCreateCityByName` already determines when a new city is created, making it the ideal place to send the message